%YAML 1.2
# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
# Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0
---
###############################
# Flatten 3D lidar data to 2D #
###############################
name: pointcloud_to_flatscan
components:
- name: tx_flatscan
  type: nvidia::gxf::DoubleBufferTransmitter
- type: nvidia::gxf::DownstreamReceptiveSchedulingTerm
  parameters:
    transmitter: tx_flatscan
    min_size: 1
- name: rx_point_cloud
  type: nvidia::gxf::DoubleBufferReceiver
- type: nvidia::gxf::MessageAvailableSchedulingTerm
  parameters:
    receiver: rx_point_cloud
    min_size: 1
- name: allocator
  type: nvidia::gxf::BlockMemoryPool
  parameters:
    storage_type: 1
    block_size: 19200000
    num_blocks: 120
- name: pointcloud_to_flatscan
  type: nvidia::isaac_ros::point_cloud::PointCloudToFlatscan
  parameters:
    allocator: allocator
    rx_point_cloud: rx_point_cloud
    tx_flatscan: tx_flatscan
    threshold_x_axis: false
    threshold_y_axis: false
    min_x: -0.0
    max_x: 4.0
    min_y: -0.0
    max_y: 4.0
    min_z: -0.1
    max_z: 0.1
    max_points: 691200
---
###########################
# Sink for ROS Publisher #
###########################
name: sink
components:
- name: signal
  type: nvidia::gxf::DoubleBufferReceiver
  parameters:
    capacity: 1
- type: nvidia::gxf::MessageAvailableSchedulingTerm
  parameters:
    receiver: signal
    min_size: 1
- name: sink
  type: nvidia::isaac_ros::MessageRelay
  parameters:
    source: signal
---
components:
- name: edge0
  type: nvidia::gxf::Connection
  parameters:
    source: pointcloud_to_flatscan/tx_flatscan
    target: sink/signal
---
components:
- name: clock
  type: nvidia::gxf::RealtimeClock
- type: nvidia::gxf::EventBasedScheduler
  parameters:
    clock: clock
    stop_on_deadlock: false
    worker_thread_number: 2
    worker_thread_name_id: "pointcloud_to_flatscan_node"
- type: nvidia::gxf::JobStatistics
  parameters:
    clock: clock
